src/ApiCheck.Task/ApiCheckGenerateTask.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.BuildTools.ApiCheck.Task
+{
+    /// <summary>
+    /// An MSBuild task that acts as a shim to <c>Microsoft.AspNetCore.BuildTools.ApiCheck.exe generate ...</c> or
+    /// <c>dotnet Microsoft.AspNetCore.BuildTools.ApiCheck.dll generate ...</c>.
+    /// </summary>
+    public class ApiCheckGenerateTask : ApiCheckTasksBase
+    {
+        public ApiCheckGenerateTask()
```

Comment:
Minor but could move explicit constructor into `ApiCheckTasksBase` and remove from `ApiCheckGenerateTask` and `ApiCheckTask`.